code/shared/game_rpc/vehicle/vehicle_setprops.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "shared/modules/vehicle_sync.hpp"
+#include "src/networking/rpc/game_rpc.h"
+#include "utils/optional.h"
+
+namespace MafiaMP::Shared::RPC {
+    class VehicleSetProps final: public Framework::Networking::RPC::IGameRPC<VehicleSetProps> {
+      public:
+        Framework::Utils::Optional<int> radioId; // -1 == turn radio off
+        Framework::Utils::Optional<Modules::VehicleSync::LockState> locked;
+        Framework::Utils::Optional<bool> beaconLights;
+        Framework::Utils::Optional<bool> siren;
+        Framework::Utils::Optional<SLNet::RakString> licensePlate;
+
+        void FromParameters(VehicleSetProps props) {
+            this->radioId      = props.radioId;
+            this->locked       = props.locked;
+            this->beaconLights = props.beaconLights;
+            this->siren        = props.siren;
+            this->licensePlate = props.licensePlate;
+        }
+
+        void Serialize(SLNet::BitStream *bs, bool write) override {
+            radioId.Serialize(bs, write);
+            radioId.Serialize(bs, write);
+            locked.Serialize(bs, write);
+            beaconLights.Serialize(bs, write);
+            siren.Serialize(bs, write);
+            licensePlate.Serialize(bs, write);
+        }
+
+        bool Valid() const override {
+            if (radioId.HasValue() && (radioId() >= -1 && radioId() < 4 /* E_RADIO_POLICE */))
```

Comment:
Is this worth being E_RADIO_LAST? However I don't actually know if E_RADIO_POLICE even does anything.

Reply:
We could test it just in case, otherwise this should be good enough. I'll get to it in a few.

Reply:
I don't think it will have any significance tbf

Reply:
Ye we can keep it as is for now I think.